admin/admin.go

```diff
@@ -0,0 +1,753 @@
+	}
+	i := rand.Intn(len(topicRoute.BrokerDataList))
+	brokerAddr := topicRoute.BrokerDataList[i].SelectBrokerAddr()
+
+	req := &internal.GetConsumerListRequestHeader{
+		ConsumerGroup: group,
+	}
+	cmd := remote.NewRemotingCommand(internal.ReqGetConsumerListByGroup, req, nil)
+	response, err := a.cli.InvokeSync(ctx, brokerAddr, cmd, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	switch response.Code {
+	case internal.ResSuccess:
+		c := &ConsumerConnection{}
+		err = json.Unmarshal(response.Body, c)
+		if err != nil {
+			rlog.Error("unmarshal consumer list info failed", map[string]interface{}{
+				rlog.LogKeyUnderlayError: err,
+			})
+		}
+		return c, nil
+	default:
+		rlog.Warning(fmt.Sprintf("unexpected response code: %d", response.Code), nil)
+	}
+	return nil, primitive.MQBrokerErr{ResponseCode: response.Code, ErrorMessage: response.Remark}
+}
+
+// GetConsumerIdList get all group client id
+func (a *admin) GetConsumerIdList(ctx context.Context, group string) ([]string, error) {
+	retryTopic := internal.GetRetryTopic(group)
+	topicRoute, _, err := a.namesrv.UpdateTopicRouteInfo(retryTopic)
+	if err != nil {
+		return nil, err
+	}
+	i := rand.Intn(len(topicRoute.BrokerDataList))
+	brokerAddr := topicRoute.BrokerDataList[i].SelectBrokerAddr()
+
+	req := &internal.GetConsumerListRequestHeader{
+		ConsumerGroup: group,
+	}
+	cmd := remote.NewRemotingCommand(internal.ReqGetConsumerListByGroup, req, nil)
+	// TODO: xujianhai666 remove timeout later
+	response, err := a.cli.InvokeSync(ctx, brokerAddr, cmd, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	switch response.Code {
+	case internal.ResSuccess:
+		result := gjson.ParseBytes(response.Body)
+		list := make([]string, 0)
+		arr := result.Get("consumerIdList").Array()
+		for idx := range arr {
+			list = append(list, arr[idx].String())
+		}
+		return list, nil
+	default:
+		rlog.Error(fmt.Sprintf("unexpected response code: %d", response.Code), nil)
+	}
+	return nil, primitive.MQBrokerErr{ResponseCode: response.Code, ErrorMessage: response.Remark}
+}
+
+// GetConsumerRunningInfo fetch running info of speified client of group
+func (a *admin) GetConsumerRunningInfo(ctx context.Context, group string, clientID string) (*internal.ConsumerRunningInfo, error) {
+	retryTopic := internal.GetRetryTopic(group)
+	topicRoute, _, err := a.namesrv.UpdateTopicRouteInfo(retryTopic)
+	if err != nil {
+		return nil, err
+	}
+	i := rand.Intn(len(topicRoute.BrokerDataList))
+	brokerAddr := topicRoute.BrokerDataList[i].SelectBrokerAddr()
+
+	req := &internal.GetConsumerRunningInfoRequestHeader{
+		ConsumerGroup: group,
+		ClientID:      clientID,
+	}
+	cmd := remote.NewRemotingCommand(internal.ReqGetConsumerRunningInfo, req, nil)
+	response, err := a.cli.InvokeSync(ctx, brokerAddr, cmd, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	switch response.Code {
+	case internal.ResSuccess:
+		info := &internal.ConsumerRunningInfo{}
+		err := info.Decode(response.Body)
```

Comment:
Could we use `Unmarshal` instead of `Decode` like in line 663? It seems after the change, test could still pass.